src/geode/model/helpers/model_component_filter.cpp

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2019 - 2022 Geode-solutions
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+#include <geode/model/helpers/model_component_filter.h>
+
+#include <geode/model/mixin/core/corner.h>
+#include <geode/model/mixin/core/line.h>
+#include <geode/model/mixin/core/surface.h>
+#include <geode/model/representation/builder/brep_builder.h>
+#include <geode/model/representation/builder/section_builder.h>
+#include <geode/model/representation/core/brep.h>
+#include <geode/model/representation/core/section.h>
+
+namespace
+{
+    template < typename Model, typename Builder >
+    void filter_surfaces( const Model& model, Builder& filter )
+    {
+        std::vector< geode::uuid > surfaces_to_delete;
+        surfaces_to_delete.reserve( model.nb_surfaces() );
+        for( const auto& surface : model.surfaces() )
+        {
+            if( model.nb_incidences( surface.id() ) == 0
+                && model.nb_embeddings( surface.id() ) == 0 )
+            {
+                surfaces_to_delete.push_back( surface.id() );
+            }
+        }
```

Comment:
You could even extract this code by taking a template Range in parameters and the number of elements to reserve and returning the vector to delete.